.github/workflows/test.yml

```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.7, 3.8]
         scikit-learn: [0.21.2, 0.22.2, 0.23.1, 0.24]
         os: [ubuntu-latest]
```

Comment:
If we keep `ubuntu-latest` here, might the same problem not happen again in the future? 
Or do we want to keep `ubuntu-latest`?

Reply:
Two things for why I am ok with this:
 1. this legacy support will be factored out into a separate scikit-learn extension module, after which only testing on `ubuntu-latest` seems fine.
 2. the correct way (I think) would be first to have some bot notification about changes to `ubuntu-latest` and then decide what to do with it based on what the new ubuntu supports